develop/commands/creating-commands.md

```diff
@@ -0,0 +1,215 @@
+
+The single method in `Command`, `run(CommandContext<S>)` takes a `CommandContext<S>` as the sole parameter and returns
+an integer. The command context holds your command source of `S` and allows you to obtain arguments, look at the parsed
+command nodes and see the input used in this command.
+
+Like other functional interfaces, it is usually used as a lambda or a method reference:
+
+```java
+Command<ServerCommandSource> command = context -> {
+    return 0;
+};
+```
+
+The integer can be considered the result of the command. Typically negative values mean a command has failed and will do
+nothing. A result of `0` means the command has passed. Positive values mean the command was successful and did
+something.
+
+#### What Can the `ServerCommandSource` Do?
+
+A `ServerCommandSource` provides some additional implementation-specific context when a command is run. This includes
+the
+ability to get the entity that executed the command, the world the command was run in or the server the command was run
+on.
+
+You can access the command source from a command context by calling `getSource()` on the `CommandContext` instance.
+
+```java
+Command<ServerCommandSource> command = context -> {
+    ServerCommandSource source = context.getSource(); 
+    return 0;
+};
+```
+
+### Registering a Basic Command
+
+Commands are registered within the `CommandRegistrationCallback` provided by the Fabric API.
+
+::: info
+For information on registering callbacks, please see the Events guide.
+:::
+
+The event should be registered in your mod's initializer.
+
+The callback has three parameters:
+
+* `CommandDispatcher<ServerCommandSource> dispatcher` - Used to register, parse and execute commands. `S`  is the type
+  of command source the command dispatcher supports.
+* `CommandRegistryAccess registryAccess` - Provides an abstraction to registries which may be passed to certain command
+  argument methods
+* `CommandManager.RegistrationEnvironment environment` - Identifies the type of server the commands are being registered
+  on.
+
+In the mod initializer, we just register a simple command:
+
+@[code lang=java transcludeWith=:::1](@/reference/latest/src/main/java/com/example/docs/command/FabricDocsReferenceCommands.java)
+
+In the `sendFeedback()` method, the first parameter is the text to be sent, which is a `Supplier<Text>` to avoid
+instantiating Text objects when not needed.
+
+The second parameter determines whether to broadcast the feedback to other
+operators. Generally, if the command is to query something without actually affecting the world, such as query the
+current time or some player's score, it should be `false`. If the command actually does something, such as changing the
+time or modifying someone's score, it should be `true`.
+
+If the command fails, instead of calling `sendFeedback()`, you may directly throw a `CommandSyntaxException`.
```

Comment:
Generally, you can throw any exception, and the server/client will handle it appropriately. `CommandSyntaxException` is usually just for syntax errors in the various arguments, nothing is stopping the developer from writing their own exception classes that the server/client can format in the chat - might be worth mentioning that too.